generic/tests/linux_stress.py

```diff
@@ -0,0 +1,208 @@
+        raise test error if not
+
+        :return True if stress process alive, otherwise False
+        """
+        session = self.vm.wait_for_login()
+        options = self.params.get('stress_options',
+                                  '--cpu 2 --io 1 --vm 1 --vm-bytes 128M')
+        try:
+            session.cmd_output_safe('stress %s &' % options)
+        # The background process sometimes does not return to
+        # terminate, if timeout, send a blank line afterward
+        except aexpect.ShellTimeoutError:
+            session.cmd_output_safe('')
+        session.close()
+        return self.check_alive()
+
+    @error_context.context_aware
+    def check_alive(self, timeout=5):
+        """
+        Check if the stress process alive or not
+
+        :param timeout: timeout to verify the process alive
+        :return: True if stress process alive, otherwise False
+        """
+        session = self.vm.wait_for_login()
+        check_alive_cmd = self.params.get('check_alive_cmd', 'pgrep stress')
+        error_context.context('Check if the stress process alive', logging.info)
+        time.sleep(timeout)
+        alive = utils_misc.wait_for(lambda:
+                                    session.cmd_output_safe(check_alive_cmd),
+                                    timeout=timeout)
+        session.close()
+        return alive
+
+    @error_context.context_aware
+    def stop_stress(self):
+        """
+        Stop the stress process, and verify if it is still alive
+        """
+        session = self.vm.wait_for_login()
+        if self.check_alive():
+            stop_command = self.params.get('stop_cmd', 'killall -g stress')
+            session.cmd_output_safe(stop_command)
+        # Log it if stop stress failed, it will not fail the test
+        # case since stop stress failure is not really harmful
+        if self.check_alive():
+            error_context.context('stop stress failed', logging.info)
+        session.close()
+
+
+def run(test, params, env):
+    """
+    General stress test for linux:
+       1). Install stress if need
+       2). Start stress process
+       3). If no stress_time defined, keep stress until test_timeout;
+       otherwise execute below steps after sleeping stress_time long
+       4). Stop stress process
+       5). Uninstall stress
+       6). Verify guest kernel crash
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    vm = env.get_vm(params['main_vm'])
+    vm.verify_alive()
+    stress = Stress(params, vm)
+    if not stress.install():
+        test.fail("stress app installation failed")
+    if not stress.launch_stress():
+        test.fail("launch stress app failed")
+    stress_time = int(params.get('stress_time'))
+    if stress_time:
+        time.sleep(stress_time)
+        stress.stop_stress()
+        stress.uninstall()
+        vm.verify_kernel_crash()
```

Comment:
@huangyum Yes, stress it untill test_timeout(default: shared/cfg/base.cfg:test_timeout = 14400). This is for when it is functioned as a subtest, users can choose kill it or not by set `stress_time` or not. And of course it can alos set up a smaller test_timeout.